numpy/random/mtrand/mtrand.pyx

```diff
@@ -641,8 +650,17 @@ cdef class RandomState:
         RandomState
 
         """
+        if version is None:
+            version = HIGHEST_VERSION if seed is None else 0
+        if version not in range(HIGHEST_VERSION + 1):
+            raise ValueError("`version` must be an integer between 0 and {}".
+                             format(HIGHEST_VERSION))
+        self.version = version
+
         cdef rk_error errcode
         cdef ndarray obj "arrayObject_obj"
+        self._free_internal_state()
+        self.internal_state = <rk_state*>PyMem_Malloc(sizeof(rk_state))
```

Comment:
This dropping and recreating internal state stuff seems superfluous to me... (and even the existence of the `_free_internal_state` method).